### ISSUE TITLE
[feature] YouTube support.

### ISSUE THREAD

**Issue by shahzeb1** [albinekb/open-pip-cli#2]
Pass in a YouTube video URL, and PIP for that video embed is activated?

Willing to fork and work on this if it sounds sensible.

**Comment by albinekb** [comment]
Sounds good! I have a version of pip.app running a webview, but it's not very good. So the better solution is to get the mp4/m4v stream from YouTube 👌

**Comment by JonnyBurger** [comment]
How about we leverage existing modules for this and allow stdin from this module? 🤓

I'm pretty sure this would work:

```sh
npm install -g ytdl

ytdl --print-url --filter-container=mp4 http://www.youtube.com/watch\?v\=_HSylqgVYQI | open-pip-cli
```

Would be happy to do a PR!

**Comment by albinekb** [comment]
I'd love that! @JonnyBurger 

PR away 👌 

Maybe it should be done in this one:
https://github.com/albinekb/open-pip

Or have it in this cli only to have the raw module "clean"

What do you think?

**Comment by JonnyBurger** [comment]
I think the CLI module should be responsible for handling `stdin`.

You are right, you don't necessarily need to pipe. I didn't think of your solution.
For bash and zsh, I had to do add a `$` to make it work:

```sh
open-pip $(ytdl --print-url --filter-container=mp4 _HSylqgVYQI)
```
Maybe my PR is still useful, because with piping the command is shell-independent.

It's very cool that we managed to hook two node CLI tools to create something new without any modifications :) 

I put `alias yt-url="ytdl --print-url --filter-container=mp4"` in my `.zshrc` to make it even easier.

**Comment by albinekb** [comment]
Released 1.2.0 with support for `stdin` from @JonnyBurger 
@shahzeb1 

you can now use `ytdl` (`npm i -g ytdl`)
```sh
open-pip $(ytdl --print-url --filter-container=mp4 _HSylqgVYQI)
```

Want to add support for just pasting an url, so leaving this issue open.

**Comment by albinekb** [comment]
Now got support for https://github.com/albinekb/open-pip-cli/releases/tag/1.3.0 youtube links! And twitch ones are documented in readme.
